src/stage1/recurse_fwd.jl

```diff
@@ -4,13 +4,51 @@ struct ∂vararg{N}; end
 
 (::∂vararg{N})() where {N} = ZeroBundle{N}(())
 function (::∂vararg{N})(a::AbstractTangentBundle{N}...) where N
-    CompositeBundle{N, Tuple{map(x->basespace(typeof(x)), a)...}}(a)
+    B = Tuple{map(x->basespace(Core.Typeof(x)), a)...}
+    return (∂☆new{N}())(B, a...)
 end
 
 struct ∂☆new{N}; end
 
-(::∂☆new{N})(B::Type, a::AbstractTangentBundle{N}...) where {N} =
-    CompositeBundle{N, B}(a)
+# we split out the 1st order derivative as a special case for performance
+# but the nth order case does also work for this
+function (::∂☆new{1})(B::Type, xs::AbstractTangentBundle{1}...)
```

Comment:
does the first order version of this actually give better performance?

Reply:
Yes. I checked with benchmark tools.
It's not a huge jump, but it is a bit better.
And given this is the most common case